src/libtiled/tilelayer.cpp

```diff
@@ -236,7 +233,10 @@ void TileLayer::setCells(int x, int y, TileLayer *layer,
                          const QRegion &mask)
 {
     // Determine the overlapping area
-    QRegion area = QRect(x, y, layer->width(), layer->height());
+    int width = qMax(layer->width(), layer->bounds().width());
+    int height = qMax(layer->height(), layer->bounds().height());
```

Comment:
I think this makes little sense, please revert it back to what it was.

Reply:
But why? Suppose a layer was created with size (1, 1) and then some cells were set out-of-bounds. If this layer is then used in `setCells` for some other layer, then technically, these 'out-of-bounds' cells should also be set in this layer. Don't you think?

Reply:
Well, the part that does not make sense, is to take the width/height of the bounds but still copy starting from 0,0.

Reply:
So I guess we need to unite `QRect(0, 0, layer->width(), layer->height()` with `layer->bounds()`.

Reply:
And adjust the offset accordingly.

Reply:
No, if you use the bounds then the `rect()` of the layer is completely irrelevant.

Reply:
I got confused by something else :confused: